compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -257,6 +259,16 @@ impl<'context> Elaborator<'context> {
             self.elaborate_global(global);
         }
 
+        // After everything is collected, we can elaborate our generated items.
+        // It may be better to inline these within `items` entirely since elaborating them
+        // all here means any globals will not see these. Inlining them completely within `items`
+        // means we must be more careful about missing any additional items that need to be already
+        // elaborated. E.g. if a new struct is created, we've already passed the code path to
+        // elaborate them.
+        if !generated_items.is_empty() {
+            self.elaborate_items(generated_items);
+        }
```

Comment:
nit, do we need this if statement? We'll fall through most of `elaborate_items` so the only thing this prevents is checking for dependency cycles.

Reply:
We do, otherwise we get infinite recursion since elaborate_items would unconditionally call itself

Reply:
Ah, I didn't see which function we were in.